src/Core/GenericEqualityComparer.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace WebLinq
+{
+    public class GenericEqualityComparer<T, U> : IEqualityComparer<U>
```

Comment:
- No need to call it generic. It's obvious so rename class to simply `EqualityComparer`
- Missing `sealed`
- `U` is unconventional